src_c/color.c

```diff
@@ -920,6 +923,60 @@ _premul_alpha(pgColorObject *color, PyObject *args)
     return (PyObject *)_color_new_internal(Py_TYPE(color), new_rgba);
 }
 
+static PyObject *
+_color_update(pgColorObject *self, PyObject *args, PyObject *kwargs)
+{
+    Uint8 *rgba = self->data;
+    PyObject *r_or_obj;
+    PyObject *g = NULL;
+    PyObject *b = NULL;
+    PyObject *a = NULL;
+
+    if (!PyArg_ParseTuple(args, "O|OOO", &r_or_obj, &g, &b, &a)) {
+        return NULL;
+    }
+
+    if (!g) {
+        if (_parse_color_from_single_object(r_or_obj, rgba)) {
+            return NULL;
+        }
+    }
+    else {
+        Uint32 color = 0;
+
+        /* Color(R,G,B[,A]) */
+        if (!_get_color(r_or_obj, &color) || color > 255) {
+            PyErr_SetString(PyExc_ValueError, "invalid color argument");
```

Comment:
Nice work on the PR :) 👍 
A small nitpick, you can use a macro to set error and return NULL, just like.
`return RAISE(PyExc_ValueError, "invalid color argument");`